client/my-sites/themes/current-theme/index.jsx

```diff
@@ -88,7 +93,13 @@ const CurrentThemeWithOptions = ( { siteId, currentTheme } ) => (
 );
 
 export default connect(
-	( state, { siteId } ) => ( {
-		currentTheme: getCurrentTheme( state, siteId )
-	} )
+	( state, { siteId } ) => {
+		const currentThemeId = getActiveTheme( state, siteId );
+		const siteIdOrWpcom = isJetpackSite( state, siteId ) ? siteId : 'wpcom';
+		return {
+			currentThemeId,
+			currentTheme: getTheme( state, siteIdOrWpcom, currentThemeId ),
```

Comment:
> We'd have to keep other selectors and their signatures in mind for consistency

Yeah that's the trade-off, so it may not make sense.

> I'd rather tackle it in a separate PR.

Absolutely